File: Frontend/src/app/models/pokemon.ts
import { Language } from "./language";
import { PokemonBase } from "./pokemon-base";
import { PokemonMove } from "./pokemon-move";

export class Pokemon {
  internalId: string;
  id: string;
  name: Language;
  type: string[];
  base: PokemonBase;
  color: string;
  move: PokemonMove;

  constructor() {

    this.internalId = null;
    this.id = null;
    this.name = null;
    this.type = [];
    this.base = null;
    this.color = null;
    this.move = new PokemonMove();
  }
}

File: Models/Pokemon.cs
using System.Collections.Generic;

using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace PokeLexApi.Models
{
    public class Pokemon
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string InternalId { get; set; }

        public string Id { get; set; }

        // [BsonElement("Language")]
        public Language Name { get; set; }

        public ICollection<string> Type { get; set; }

        public PokemonBase Base { get; set; }
    }
}